tendrl/ceph_integration/objects/rbd/atoms/rbd_not_exists/__init__.py

```diff
@@ -12,29 +12,32 @@ def __init__(self, *args, **kwargs):
         super(RbdNotExists, self).__init__(*args, **kwargs)
 
     def run(self):
-        Event(
-            Message(
-                priority="info",
-                publisher=NS.publisher_id,
-                payload={
-                    "message": "Checking if rbd: %s doesnt exist for pool %s" %
-                    (self.parameters['Rbd.name'],
-                     self.parameters['Rbd.pool_id'])
-                },
-                job_id=self.parameters['job_id'],
-                flow_id=self.parameters['flow_id'],
-                cluster_id=NS.tendrl_context.integration_id,
-            )
-        )
-        try:
-            NS.etcd_orm.client.read(
-                'clusters/%s/Pools/%s/Rbds/%s' % (
-                    NS.tendrl_context.integration_id,
-                    self.parameters['Rbd.pool_id'],
-                    self.parameters['Rbd.name']
+        if self.parameters.get('Rbd.pool_id') is not None:
+            Event(
+                Message(
+                    priority="info",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Checking if rbd: %s doesnt exist for pool"
+                                   " %s" % (self.parameters['Rbd.name'],
+                                            self.parameters['Rbd.pool_id'])
+                    },
+                    job_id=self.parameters['job_id'],
+                    flow_id=self.parameters['flow_id'],
+                    cluster_id=NS.tendrl_context.integration_id,
                 )
             )
-        except etcd.EtcdKeyNotFound:
+            try:
+                NS.etcd_orm.client.read(
```

Comment:
create an actual Tendrl object(https://github.com/Tendrl/ceph-integration/blob/develop/tendrl/ceph_integration/objects/rbd/__init__.py)

 Rbd(pool_id, name).exists() , dont use NS.etcd_orm.client, it is going to be deprecated

https://github.com/Tendrl/commons/blob/develop/tendrl/commons/objects/__init__.py#L120